src/plugins/kernel_install/utils.sh

```diff
@@ -550,32 +550,85 @@ function do_uninstall()
   fi
 }
 
+# Element existence check
+# Checks if element belongs to given array
+#
+# @target_element element being checked
+# @array target array
+# returns 1 if element is in array and 0 otherwise
+function is_in_array()
+{
+  local target_element="$1"
+  local array=("${!2}")
+
+  for element in "${array[@]}"; do
+    if [[ "$element" == "$target_element" ]]; then
+      return 1
+    fi
+  done
+  return 0
+}
+
+# Return available kernels
+#
+# @prefix Set a base prefix for searching for kernels.
+# @all_kernels List all available kernels, not only the ones installed by kw
+# returns array containing available kernels
+function process_installed_kernels()
+{
+  local all_kernels="$1"
+  local prefix="$2"
+  local kernels
+
+  kernels=$(list_installed_kernels 'SILENT' 1 "$all_kernels" "$prefix")
+  IFS=, read -r -a available_kernels <<< "$kernels"
+  mapfile -t deduped_kernels < <(printf "%s\n" "${available_kernels[@]}" | sort --unique)
```

Comment:
Following the comment above, here you will use `mapfile -t _processed_installed_kernels ...`.

Also, IMHO, we can refine the right part of this line to be cleaner by making it
```bash
mapfile -t _processed_installed_kernels <<< "$(printf "%s\n" "${available_kernels[@]}" | sort --unique)"
```